core/arch/x86/decode_table.c

```diff
@@ -3679,16 +3709,15 @@ const instr_info_t prefix_extensions[][12] = {
     {OP_movd,   0x0f7e10, "movd", Ed_q, xx, Pd_q, xx, xx, mrm, x, tpe[51][2]},
     /* movq zeroes the top bits when the destination is an mmx or xmm reg */
     {OP_movq, 0xf30f7e10, "movq", Vdq, xx, Wq_dq, xx, xx, mrm, x, tpe[61][2]},
-    {OP_movd, 0x660f7e10, "movd", Ed_q, xx, Vd_dq, xx, xx, mrm, x, END_LIST},
```

Comment:
Not sure about this: isn't this 4 bytes w/o rex.w, and 8 bytes w/ rex.w?  I.e., the size looks correct at first glance to me (though its name `d_q` seems to violate the conventions: shouldn't be an underscore...separate issue).

As mentioned earlier, it seems best to separate things like this out from adding the new opcodes.

Reply:
Ok, looks like that we're calling both the 66 0F 7E /r and the  66 REX.W 0F 7E /r version "movd", while Intel calls the REX.W version "movq". That must have confused me, but you're right it should be Ed_q - for now. I've reverted this. Thanks! File an issue to change into separate entry + test?

Reply:
There are numerous cases where Intel names an opcode differently but we don't bother to split our table entries (see "Alternative opcode name" in opcode.h) so I'm not sure we need to do this.  For first-class disassembly support we might, but then we'd also want to go split all those others (cdge, cqo, jrcxz, cmpxchg16b, etc.).  We already have #1307 for them so this movd/movq would fall under that issue.

Reply:
Ok, see comment below.